files/common/usr/bin/unpack-image

```diff
@@ -42,15 +42,15 @@ function cleanup() {
 # the same version. Otherwise, this will fail if there's already an
 # unpacked image of the same version.
 #
-opt_f=false
+opt_f=""
```

Comment:
Let's just remove the declaration of these options, and their comments. I see you copied the comments over to the new `prepare` script, and I think that's the more appropriate place for them. And these declarations don't do anything special, so I think removing them would be best (the script will work the same).

Reply:
Can remove the comments but since this is the script that any customer orchestration would actually use, I believe we should leave it in there. Unless of course if these options are meant to be hidden (I do not get why the -s option should ever be exposed in production version of this script). But removing the comments would not effectively achieve hiding the options either so I am inclined to leave it in here. If you feel they're redundant in two scripts, I vote for removing the comments in the new prepare script instead. What do you think?

The shell format check complains about using uninitialized variables when using these when calling the prepare script, so will need to keep the initializations I believe.

Reply:
We can suppress the static analysis error. I still vote to remove the declarations and the suppress the error, but if you disagree, I won’t push for it; I’m OK either way.

And yea, this script isn’t exposed to customers, but at the same time the changes I’m suggesting wouldn’t change the external interface of this script, so I don’t think it’d matter even if we did expose this to customers.

Reply:
Cool, I have removed the comments. I do like the initializations just for readability since the code where these variables are constructed do not make them obvious.

Reply:
Hmm just realized I am only using the `-s` option in the unpack-image script at the moment. I will continue to pass it into prepare since we are not yet sure if there will be necessity for signature checks within the prepare script for any additional artifacts like the verification package (there shouldn't be. The single SHA256SUMS file should in theory have the hashes for all artifacts and this file will be signed but I would rather pass in more info than less in case we need this param for our test environment).

In any case, given the parameter is used in unpack-image, I am considering moving the comments over from prepare to unpack-image instead.

Reply:
Also realizing an empty string does not evaluate to false in bash. I am going to need separate variables to pass along to prepare like I had before.

Reply:
Checking for empty string removes need for separate variable. Using that in latest upload.

Reply:
I don't think we should pass in `-s` if it's not used by `prepare`. I also don't think `prepare` should accept `-s` if it doesn't use that option.

I can easily imagine `prepare` being modified later to remove `-s`, due to somebody noticing it's not used in `prepare`, but not realizing `unpack-image` uses it for no reason.

Reply:
The concern here is with upgrading from Lumen to a future prepare script created in SUV to handle unpacking of the new meta-tar based upgrade image. At the moment I do not believe we have sufficient information to decide if this parameter will NOT be needed by that script.

Reply:
Ill think a bit more about this and our plans for the build process in SUV and update the branch.

Reply:
> At the moment I do not believe we have sufficient information to decide if this parameter will NOT be needed by that script.

But this can be said about any possible option that can potentially be passed in. Based on this argument, we should accept all letters of the alphabet, and pass them to `prepare`, even if they're not used today.

Further, I don't think `-s` is even used by the product. IIRC, it's only there for development purposes, right? (I could be mistaken, I haven't reviewed the product code to verify this claim)

Reply:
I think there is a misunderstanding with my statement. The concern is with respect to parameters being accepted by the lower version script which is a definite set we know of. To pass in all of these known set of parameters is all I am considering, given the fact that we are in effect splitting one script into two. Regarding the specific parameter -s, I too do not know where it is used but it is not passed in from the upgrade api that handles upload of the upgrade image. That code path only passes in the image file name so all the parameters are for internal use only, potentially in our qa tests.

Reply:
Perhaps I took it too far in my prior comment...

So, thinking a little more broadly, are you suggesting we adopt the design that `unpack-image` and `prepare` must maintain the same interface, and that `unpack-image` must communicate to `prepare` all of the options specified to it when it was executed?

My issue with this design, is `prepare` will need to work with multiple different versions of `unpack-image`. So, I can imagine a situation where we remove `-s` from `unpack-image` in some later release, or we change the meaning of `-s`.. I'm not sure how `prepare` would work in this situation, since `prepare` would have to match the interface of two different versions of `unpack-image`.

Perhaps we could make this work, though, since `prepare` will know the version of `unpack-image` being used, so it could interpret the options differently based on the version of `unpack-image` that's calling it?

If you're only suggesting that `prepare` accepts `-s`, without any broader design goals, I find it difficult to justify this, since `-s` has no meaning to `prepare` right now (and we have no way to test and verify `-s` works properly).

Ultimately, I'm OK with whatever approach you want to take here. I may not fully agree, but I don't think it's worth blocking this PR over it either. If you want to pass `-s` from `unpack-image` to `prepare`, that's fine with me.

Reply:
> So, thinking a little more broadly, are you suggesting we adopt the design that unpack-image and prepare must maintain the same interface, and that unpack-image must communicate to prepare all of the options specified to it when it was executed?

Correct. The design reason is that at the time a particular version is released, all parameters used in its unpack-image script are deemed necessary. It would be reasonable to pass it along to the prepare script from a newer release and let that script handle the parameter correctly.

> My issue with this design, is prepare will need to work with multiple different versions of unpack-image. So, I can imagine a situation where we remove -s from unpack-image in some later release, or we change the meaning of -s.. I'm not sure how prepare would work in this situation, since prepare would have to match the interface of two different versions of unpack-image.

Good point. This is inherent in the design of SUV that the `prepare` script in SUV will handle a relatively complex matrix of upgrades since it will be a single script delphix-wide. So parameters will be just one aspect among many that could be treated in a conditional manner based on the current_version. Having said that, I do not expect the list of parameters to expand too large and neither to change too often so I think we are fine in that respect.

By removing a parameter, I am worried we might be handicapping ourselves when the time comes to write and ship the new SUV script. Since there are no security/performance concerns with this, let's pass in this parameter at this time.

Reply:
OK. I think this is making more sense to me now.

Can you document this design someplace? Maybe a comment in the “prepare” script?